app/views/recipes/new.html.erb

```diff
@@ -0,0 +1,14 @@
+<h1>I'm here to create a new recipe</h1>
+
+  <%= form_for :recipe, url: user_recipes_path  do |f| %>
```

Comment:
If you're using `form_for` you usually don't need to explicitly set the URL. Rails can derive it from the object that you've given it e.g. `form_for @recipe do |f|`.

Since your recipe route is nested under a user, you'll probably want to do something like:

```ruby
form_for [@user, @recipe] do |f|
```

Reply:
I was going to link to some docs here and it looks like `form_for` has been [deprecated since Rails 5.1](https://guides.rubyonrails.org/form_helpers.html#using-form-for-and-form-tag)

> Before `form_with` was introduced in Rails 5.1 its functionality used to be split between `form_tag` and `form_for`. Both are now soft-deprecated. Documentation on their usage can be found in older versions of this guide.

The behavior for [`form_with`](https://api.rubyonrails.org/v6.0.1/classes/ActionView/Helpers/FormHelper.html#method-i-form_with-label-Resource-oriented+style) is similar to what I have shown earlier (note the addition of the `model:` key):

```ruby
form_with model: [@user, @recipe] do |f|
end
```

The docs show [some examples](https://api.rubyonrails.org/v6.0.1/classes/ActionView/Helpers/FormHelper.html#method-i-form_with-label-Examples) of using this style for nested resources.

Reply:
Thanks for this! I will shift out of using form_for.